nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -378,6 +397,36 @@ void CollisionMonitor::process(const Velocity & cmd_vel_in)
     }
   }
 
+  if (collision_points_marker_pub_) {
+    // visualize collision points with markers
+    visualization_msgs::msg::MarkerArray marker_array;
+    if (collision_points.size() > 0) {
+      visualization_msgs::msg::Marker marker;
+      marker.header.frame_id = get_parameter("base_frame_id").as_string();
+      marker.header.stamp = rclcpp::Time(0, 0);
```

Comment:
... `now()`?

Reply:
Tried that, for some reason it breaks rviz, which complains about the TF map->baselink requiring extrapolation in the future. My `get_parameter("base_frame_id").as_string()` is `base_link` and my Rviz fixed frame is `map`.

I guess the current implementation is also fine according to [this](https://wiki.ros.org/rviz/DisplayTypes/Marker#:~:text=Note%20that%20the,the%20current%20time.)

Reply:
OK, its just visualization anyway